### PR TITLE
[sql lab] fix table dropdown with large schema make UI unresponsive

### PR DESCRIPTION
**Pull request by mistercrunch**
`createFilterOptions` which I assume does some sort of indexing, was done on render which is no good. Moving to doing it only when the data changes.

Also I fixed a few React warnings while I was testing things.

@ascott @jefffeng

### CODE REVIEW

**Comment by mistercrunch** [comment]
+ @williaster

File: superset/assets/javascripts/SqlLab/components/SqlEditor.jsx [hunk 1]
```diff
@@ -148,7 +148,7 @@ class SqlEditor extends React.PureComponent {
         <div className="pull-left">
           <Form inline>
             <RunQueryActionButton
-              allowAsync={this.props.database && this.props.database.allow_run_async}
+              allowAsync={this.props.database ? this.props.database.allow_run_async : false}
```
Comment (williaster) [inline superset/assets/javascripts/SqlLab/components/SqlEditor.jsx]: [not that important] I'm surprised this doesn't cause an `eslint` error, was it complaining about not being a boolean? `!!this.props.database && this.props.database.allow_run_async` or `Boolean(this.props.database) && this.props.database.allow_run_async` could be alternatives.

**Review by williaster — APPROVED** [review]
this LGTM for the performance gains in render.

File: superset/assets/javascripts/SqlLab/components/SqlEditor.jsx (hunk 1 above)
Comment (mistercrunch) [inline superset/assets/javascripts/SqlLab/components/SqlEditor.jsx]: We use this in a fair amount of places. The casting to boolean is implicit. I know the equivalent is considered pythonesque in python but not sure about JS...

**Comment by ascott** [comment]
thx @mistercrunch! 🎉

**Pull request merged**